certs/certs.go

```diff
@@ -34,6 +35,60 @@ import (
 	"sigs.k8s.io/etcdadm/constants"
 )
 
+// GetDefaultCertList returns all of the certificates etcdadm requires to function.
+func GetDefaultCertList() []string {
+	return []string{
+		//constants.EtcdCACertAndKeyBaseName,
+		constants.EtcdServerCertAndKeyBaseName,
+		constants.EtcdPeerCertAndKeyBaseName,
+		constants.EtcdctlClientCertAndKeyBaseName,
+		constants.APIServerEtcdClientCertAndKeyBaseName,
+	}
+}
+
+// RenewUsingLocalCA executes certificate renewal using local certificate authorities for generating new certs.
+func RenewUsingLocalCA(pkiDir, name string) (bool, error) {
+	// reads the current certificate
+	cert, err := pkiutil.TryLoadCertFromDiskIgnoreExpirationDate(pkiDir, name)
+	if err != nil {
+		return false, err
+	}
+
+	// extract the certificate config
+	cfg := certToConfig(cert)
```

Comment:
For `etcdadm certs renew` this means that the source of truth is the certificate on disk.. However, for `etcdadm init` and `etcdadm join`, the source of truth is [etcdadm](https://github.com/kubernetes-sigs/etcdadm/blob/bec4908b5008104cf1637644335a6b0c986bcf5a/certs/certs.go#L190).

I think that the source of truth should be consistent across all commands.

In fact, if  `etcdadm certs renew` uses etcdadm as the source of truth, then it can just create new certificates using the existing functions (`CreateEtcdServerCertAndKeyFiles`, `CreateEtcdPeerCertAndKeyFiles`,  `CreateEtcdctlClientCertAndKeyFiles`, and `CreateAPIServerEtcdClientCertAndKeyFiles`).

WDYT?

Reply:
You mean that maybe the disk certificates not the source of truth? 

I understand your said, But if we use createXXX to recreate the new certificates, is it include CA certificate?

Reply:
> You mean that maybe the disk certificates not the source of truth?

That's right; I believe that the certificates on disk should not be the source of truth.

When etcdadm creates certificates the first time, it determines their properties. When it renews them, it should set those properties the same way. It should not pay attention to the properties in the certificates on disk.

> I understand your said, But if we use createXXX to recreate the new certificates, is it include CA certificate?

I believe each of the functions I mentioned reads the CA from disk.

Reply:
Do not believe the certificates on disk is right, but if we recreate certificates, this command name still call `renew`, or change to `etcdadm certs recreate` ?

Reply:
Great question! I think `renew` is correct, based on the RFC:

> This subcomponent is used to describe the following elements related to certificate renewal. Certificate renewal means the issuance of a new certificate to the subscriber without changing the subscriber or other participant's public key or any other information in the certificate:
-- https://tools.ietf.org/html/rfc3647 *

However, if I'm reading the definition correctly, etcdadm must generate a new certificate using its current key on disk. That means etcdadm needs to read the key from disk, but not the certificate.

(I continue to believe that etcdadm should set the certificate properties, rather than read them from the certificate on disk.)

\* I found this via https://security.stackexchange.com/questions/150907/x-509-certificate-renew-vs-rekey

Reply:
I think this renew implementation more like rekey, but it also use certificate  properties. If my understanding correct, you mean that etcdadm not use old certificate properties on the disk?

Reply:
I agree; this PR right now implements "rekey" according to the RFC.

I don't know whether we want "renew" or "rekey;" I need to learn the trade-offs and use cases.

I decided to check what kubeadm does. Looks like kubeadm does in fact read the properties of the existing certificates:
> Note: alpha certs renew uses the existing certificates as the authoritative source for attributes (Common Name, Organization, SAN, etc.) instead of the kubeadm-config ConfigMap. It is strongly recommended to keep them both in sync.
-- https://v1-17.docs.kubernetes.io/docs/tasks/administer-cluster/kubeadm/kubeadm-certs/#renew-certificates-with-the-kubernetes-certificates-api

I wonder why that is.

Reply:
I see kubeadm read certificate properties, i think maybe etcdadm init or join can add custom SANs, if we don't read properties, maybe we missing some SANs.

Reply:
>  i think maybe etcdadm/kubeadm init or join can add custom SANs, if we don't read properties, maybe we missing some SANs.

That's a great point.